app/questionnaire/placeholder_transforms.py

```diff
@@ -80,11 +81,27 @@ def format_number(number):
 
         return ''
 
-    def calculate_years_difference(self, first_date, second_date):
-        first_date = datetime.now() if first_date == 'now' else datetime.strptime(first_date, self.input_date_format)
-        second_date = datetime.now() if second_date == 'now' else datetime.strptime(second_date, self.input_date_format)
+    @staticmethod
+    def calculate_years_difference(first_date, second_date):
+        return str(relativedelta(PlaceholderTransforms.parse_date(second_date),
+                                 PlaceholderTransforms.parse_date(first_date)).years)
+
+    @staticmethod
+    def parse_date(date):
+        """
+        :param date: string representing a date
+        :return: datetime of that date
+
+        Convert `date` from string into `datetime` object. `date` can be 'YYYY-MM-DD', 'YYYY-MM'
+        or 'now'. Note that in the shorthand YYYY-MM format, day_of_month is assumed to be 1.
+        """
+        if date == 'now':
+            return datetime.now()
 
-        return str(relativedelta(second_date, first_date).years)
+        try:
+            return datetime.strptime(date, PlaceholderTransforms.input_date_format)
+        except ValueError:
+            return datetime.strptime(date, PlaceholderTransforms.input_date_format_md_only)
```

Comment:
I wonder if inspecting the input date string and checking if it matches a known format would be better than falling back here?

Reply:
As before/later (depending on the order in which the comments are read), I wasn't wanting to tread on anyone's toes and change the way it worked more than I had to. I had considered a regexp, or a count of dashes. I was surprised there wasn't an equivalent to a tryParse() method. You have any preference?

Reply:
I should think whether it matches a regex for either date format would be enough.

Reply:
Consensus seems to be that it isn't necessary, but I won't be offended if you add that in.